mlflow/tensorflow/__init__.py

```diff
@@ -459,26 +590,7 @@ def _load_tensorflow_saved_model(tf_saved_model_dir, tf_meta_graph_tags, tf_sign
     return loaded_sig[tf_signature_def_key]
 
 
-def _parse_flavor_configuration(flavor_conf, model_path):
-    """
-    :param path: Local filesystem path to the MLflow Model with the ``tensorflow`` flavor.
-    :return: A triple containing the following elements:
-
-             - ``tf_saved_model_dir``: The local filesystem path to the underlying TensorFlow
-                                       SavedModel directory.
-             - ``tf_meta_graph_tags``: A list of tags identifying the TensorFlow model's metagraph
-                                       within the serialized ``SavedModel`` object.
-             - ``tf_signature_def_key``: A string identifying the input/output signature associated
-                                         with the model. This is a key within the serialized
-                                         ``SavedModel``'s signature definition mapping.
-    """
-    tf_saved_model_dir = os.path.join(model_path, flavor_conf["saved_model_dir"])
-    tf_meta_graph_tags = flavor_conf["meta_graph_tags"]
-    tf_signature_def_key = flavor_conf["signature_def_key"]
-    return tf_saved_model_dir, tf_meta_graph_tags, tf_signature_def_key
-
-
-def _load_pyfunc(path):
+def _load_pyfunc(path, *, model_type):
```

Comment:
Instead of adding `model_type` to the function signature and computing the model type in `mlflow.pyfunc.load_model()`, can we instead read the MLModel conf file from the directory specified by `path` and then call `_infer_model_type(model_conf)` on it?

Reply:
This is the way I tried but I hit an issue:

When loading Keras model, the `path` is the `model/data` path (Note the MLModel conf file is in `model` path)
When loading saved TF estimator model, the `path` is the `model`.

So it makes me hard to read the MLModel conf file, at this point I don't know whether it is keras or TF estimator model.

Reply:
Got it. Can we check for `MLModel` in the current directory and the directory above? It has to be one of these.

Reply:
Updated. Adopted `check for MLModel in the current directory and the directory above`